backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeBoundsProvider.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Thales.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.diagrams.components;
+
+import java.util.List;
+
+import org.eclipse.sirius.web.annotations.graphql.GraphQLNonNull;
+import org.eclipse.sirius.web.components.Element;
+import org.eclipse.sirius.web.diagrams.Position;
+import org.eclipse.sirius.web.diagrams.Size;
+
+/**
+ * Provides the bounds (Position and Size) to apply to a new node.
+ *
+ * @author fbarbin
+ */
+public class NodeBoundsProvider {
```

Comment:
The ADR is quite clear, you need a `NodePositionProvider` and a `NodeSizeProvider`

Reply:
I thought that those two informations (Size en Position) could have been merged into one Bounds provider.

Reply:
We are using ADR to communicate with our community and we have informed then that we would use two different providers so unless we have an issue, we will stick with this plan. On top of that both problems (size and position) will evolve with complex algorithms so I'm not sure that we want to mix them.